### PR TITLE
[depends] bump gnutls 3.6.14 and libgpg-error  for Leia

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

supersedes https://github.com/xbmc/xbmc/pull/18179

## Motivation and Context

fixes a severe security issue in gnutls
fixes a build error on some linux distros

## How Has This Been Tested?

build on linux, installed and ran a few addons

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/libgpg-error/Makefile [hunk 1]
```diff
@@ -22,14 +24,14 @@ $(TARBALLS_LOCATION)/$(ARCHIVE):
 $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+	cd $(PLATFORM); patch -p1 -i ../libgpg-error-1.36-gawk5-support.patch
+ifeq ($(TARGET_PLATFORM),appletvos)
+	cd $(PLATFORM); patch -p1 -i ../tvos_remove-fork.patch
+endif
```
Comment (fuzzard) [inline tools/depends/target/libgpg-error/Makefile:30]: Apple tv doesn't exist in Leia, no need for the patch

**Comment by DaveTBlake** [comment]
Partial backport of https://github.com/xbmc/xbmc/pull/18178

What is happening with iOS?

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
Nice greens, I assume ready to go @wsnipex @phunkyfish ?

**Comment by wsnipex** [comment]
I'd appreciate some testing. Installing a couple addons, stream some https URLs via ffmpeg

**Comment by phunkyfish** [comment]
Mac build is good.

**Comment by AlwinEsch** [comment]
Have tested the Linux build and was for creation OK.

Only not becomed it working for repository and addon download, but seems not related to here.
Has used the whole depends build to check about this request, only has it used for `./tools/depends/target/dbus` his test build prefix and bring:
```
DBus connection failed: org.freedesktop.DBus.Error.FileNotFound - Failed to connect to socket /home/alwin/123/x86_64-linux-gnu-debug/var/run/dbus/syst
em_bus_socket: No such file or directory
```

Also if I symlink to correct, comes:
```
2020-07-23 04:24:21.917 T:2712244   DEBUG <general>: CurlFile::Open(0x7f19a1811080) http://mirrors.kodi.tv/addons/matrix/addons.xml.gz
2020-07-23 04:24:22.046 T:2712244   ERROR <general>: CCurlFile::FillBuffer - Failed: SSL peer certificate or SSH remote key was not OK(60)
2020-07-23 04:24:22.046 T:2712244   DEBUG <general>: CCurlFile::Open - effective URL: <https://ftp.halifax.rwth-aachen.de/xbmc/addons/matrix/addons.xml.gz>
2020-07-23 04:24:22.046 T:2712244   ERROR <general>: CRepository: http://mirrors.kodi.tv/addons/matrix/addons.xml.gz index has wrong digest e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b85
5, expected: 101dce3f4d75301a4d7c14b324dbd460857906c3919e861340555e8a92de474b
```

This is also on Kodi Matrix and think this errors are not related to request here.

A crosscheck of old Leia was not possible, as the `libgpg-error` build fails there (with this request build OK)

**Comment by AlwinEsch** [comment]
What has me confused a bit was that in Kodi's depends build (Leia and Matrix) few parts was as shared library:
- libasound.so.2.0.0 (only as shared present)
- libass.so.5.3.2 (also the static present libass.a)
- libbluray.so.2.0.2 (also the static present libbluray.a)
- libinput.so.10.13.0 (only as shared present
- libsmbclient.so.0 (only as shared present)
- libshairplay.so (also the static present libshairplay.a)
- libplist.so (also the static present libplist++.a)
- libcec.so (only as shared present)
- libmariadb.so.3 (as static is libmariadbclient.a present, not sure if interact)

About this was the include in lib folder:
- /x86_64-linux-gnu-debug/lib/dbus-1.0/include/dbus
- /x86_64-linux-gnu-debug/lib/libffi-3.2.1/include
- /x86_64-linux-gnu-debug/lib/libzip/include

**Comment by AlwinEsch** [comment]
Seen one part where seems bad with this request!

The static `libgpg-error.a` was created by depends build, but in created kodi-x11 was `libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0` used,

**Comment by wsnipex** [comment]
running depends builds on Linux has all kinds of issues when the install prefix is not /use and there are system libraries around. Nobody uses this in the real world and it should be treated as build test. You can get around the mirror issue by setting a proper SSL cacerts via SSL_CERT_FILE env var

**Comment by DaveTBlake** [comment]
Dam, caused a rebase with MariaDB stuff. @wsnipex / @phunkyfish  can you fix?

**Comment by DaveTBlake** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
WIN UWP seems broken.
13:44:30 C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" exited with code -1073741515. [C:\Users\Kodi\Kodi\jenkins\builds\workspace\WIN-UWP-64\kodi-build\build\interfaces\json-rpc\schema\generate_json_header.vcxproj]

this is totally unrelated to this PR

**Comment by DaveTBlake** [comment]
Last build UWP was fine, but OSX failed - jenkins glitches?
jenkins build this please

**Comment by wsnipex** [comment]
UWP failed with the same error...

**Comment by DaveTBlake** [comment]
The nightly build ran OK on a different server, we have had server specific build issues with UWP before. I wonder if there is a way to force a build on  Win-SKYNET-Compiler rather than Win-Montellese?

**Comment by DaveTBlake** [comment]
jenikins build this please

**Comment by DaveTBlake** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
If UWP fails again it’s probably ok at this point to note that the UWP failure is unrelated and proceed.

**Pull request merged**